api/samples/neoncounter.c

```diff
@@ -0,0 +1,599 @@
+           (count_arith / (long double)count_all));
+    printf("  BRANCHING              %lu / %Lf  \n", count_branching,
+           (count_branching / (long double)count_all));
+    printf("      |___ TAKEN             %lu / %Lf  \n", count_taken_branches,
+           (count_taken_branches / (long double)count_branching));
+    printf("  LOADING                %lu / %Lf  \n", count_load,
+           (count_load / (long double)count_all));
+    printf("      |___ LINEAR            %lu / %Lf  \n", count_load_linear,
+           (count_load_linear / (long double)count_load));
+    printf("      |___ STRUCTURED        %lu / %Lf  \n", count_load_structured,
+           (count_load_structured / (long double)count_load));
+    printf("  STORING                %lu / %Lf  \n", count_store,
+           (count_store / (long double)count_all));
+    printf("      |___ LINEAR            %lu / %Lf  \n", count_store_linear,
+           (count_store_linear / (long double)count_store));
+    printf("      |___ STRUCTURED        %lu / %Lf  \n", count_store_structured,
+           (count_store_structured / (long double)count_store));
+    printf("  OTHER                  %lu / %Lf  \n",
+           count_all - (count_arith + count_load + count_store + count_branching),
+           ((count_all - (count_arith + count_load + count_store + count_branching)) /
+            (long double)count_all));
+
+    printf("=========================================================== RESULTS "
+           "===\n");
+
+    // print results in file
+    fprintf(result_file, "%s ", executable);
+    fprintf(result_file, "%lu ", count_all);
+    fprintf(result_file, "%lu ", count_simd);
+    fprintf(result_file, "%lu ", count_simd_arith);
+    fprintf(result_file, "%lu ", count_simd_load);
+    fprintf(result_file, "%lu ", count_simd_store);
+    fprintf(result_file, "%lu ", count_arith);
+    fprintf(result_file, "%lu ", count_branching);
+    fprintf(result_file, "%lu ", count_taken_branches);
+    fprintf(result_file, "%lu ", count_load);
+    fprintf(result_file, "%lu ", count_load_structured);
+    fprintf(result_file, "%lu ", count_load_linear);
+    fprintf(result_file, "%lu ", count_store);
+    fprintf(result_file, "%lu ", count_store_linear);
+    fprintf(result_file, "%lu ", count_store_structured);
+    fprintf(result_file, "%lu\n",
+            count_all - count_arith - count_load - count_store - count_branching);
+
+    fclose(result_file);
+
+    // Unregister event
+    if (!drmgr_unregister_bb_insertion_event(event_app_instruction)) {
+        DR_ASSERT(false);
+    }
+    drmgr_exit();
+}
+
+DR_EXPORT void
+dr_client_main(client_id_t id, int argc, const char *argv[])
+{
+    // set client name
+    dr_set_client_name("DynamoRIO Sample Client 'neoncounter'",
+                       "http://dynamorio.org/issues");
+    disassemble_set_syntax(DR_DISASM_ARM);
+
+    result_file = fopen("nc_output.txt", "a");
```

Comment:
This would be better as an option that specifies the path, rather than hardcoded to a file in the current directory.  This tool could be run from a read-only directory, e.g., or the user may want control over where this goes.